Backend/app/api/dialogue.py

```
@@ -62,7 +64,12 @@ def should_use_rag(user_input: str) -> bool:
     
     return True
 
-async def generate_rag_enhanced_response(user_input: str, character_id: str, scene_context: str) -> str:
+async def generate_rag_enhanced_response_with_memory(
+    user_input: str, 
+    character_id: str, 
+    scene_context: str, 
+    session: 'ConversationSession'
```

Ruff (0.12.2)

Undefined name `ConversationSession`

(F821)